include/dali/core/tensor_shape.h

```diff
@@ -253,11 +246,33 @@ template <int ndim>
 struct TensorShape : public TensorShapeBase<DeviceArray<int64_t, ndim>, ndim> {
   using Base = TensorShapeBase<DeviceArray<int64_t, ndim>, ndim>;
 
-  TensorShape(const std::array<int64_t, ndim> &s) : Base(s) {}  // NOLINT
-  DALI_HOST_DEV TensorShape(const DeviceArray<int64_t, ndim> &s) : Base(s) {}  // NOLINT
+  TensorShape(const std::array<int64_t, ndim> &s)  // NOLINT(runtime/explicit)
+      : Base(s) {}
+
+  DALI_HOST_DEV TensorShape(const DeviceArray<int64_t, ndim> &s)  // NOLINT(runtime/explicit)
+      : Base(s) {}
+
+  template <typename It,
+            typename = std::enable_if<is_integer_iterator<It>::value>>
+  TensorShape(It first, It last) : Base() {
```

Comment:
missing:
DALI_NO_EXEC_CHECK
DALI_HOST_DEV
Unnecessary explicit invocation of base's default constructor.